### PR TITLE
Smarty - Remove "bonkers" way of fetching help titles

### PR DESCRIPTION
**Pull request by colemanw**
Background
-------
Popup help text is loaded by the `CRM.help()` js function. It takes a fileName and fieldID, and fetches help text from the given `.hlp` file via ajax. Note that it does *not* fetch the title. The optional title must be passed into the function, or a title will not be shown (which is fine actually, the help text is what's important).

The smarty `{help}` function generates markup for a link which calls `CRM.help()` on-click. In order to pass a "title" to that function, it has 3 ways of figuring it out:

1. **Explicit**: `$title` can be passed directly in like `{capture assign='helpTitle'}{ts)First Name{/ts}{/capture}{help id="first_name" title=$helpTitle}`. Simple but verbose.
2. **DRY**: If `fieldID` matches the name of a field on the form, it can infer title from the field's textLabel. This is the most elegant but alas was broken. As of 33d199dd it was checking a variable named `$vars` which had not been defined yet.
3. **Bonkers**: This method was very complex and, according to the docblock, "a bit bonkers." It would fetch, load, and parse the `.hlp` file, just to get the title which is explicitly written there using the magic string concatenation of `-title` onto the $fieldID.

The last method is IMO not worth supporting. It's even more verbose than method 1, and way more complex than necessary. 

This Change
---------
This PR removes support for method 3, and restores method 2 to working order.

It updates every `.hlp` file to remove all magic `-title`s in favor of method 1 or 2.

In most cases, I found that method 2 worked. However some help links did not have an associated form element, so then I fell back onto method 1.

Comments
-------
Note: This change affects Smarty pages/forms only. JS/Angular callers of `CRM.help` are unaffected.

### CODE REVIEW

**Comment by demeritcowboy** [comment]
You bring up a good point - I mean, do we EVER need a title?

**Comment by eileenmcnaughton** [comment]
I see bonkers is the official term

image

**Comment by demeritcowboy** [comment]
Internal machine name: "bonkers"
Public title: "Colourful"

**Comment by colemanw** [comment]
@demeritcowboy the titles may help with accessibility as they're also used in the `aria-label` of the tag. I'm working on a script that will fix them all...

**Comment by colemanw** [comment]
@demeritcowboy @eileenmcnaughton this has been a days-long slog but I've finally converted all the `.hlp` files to the new format. Encountered the usual number of dragons along the way and fixed a bunch of help that was previously broken. This is ready to merge.

**Comment by demeritcowboy** [comment]
Spot-checked a couple and checked extensions and overrides. Since this is all just help text I figure the worst that will happen is people will be confused about how to use civi. The rest of that joke writes itself.

**Comment by colemanw** [comment]
Thanks @demeritcowboy I feel a little saner now.
